File: NET_8/FoodCall.Domain/Exceptions/BusinessRuleException.cs
namespace FoodCall.Domain.Exceptions;

public class BusinessRuleException : DomainException
{
    public BusinessRuleException(string message) 
        : base(message)
    {
    }

    public BusinessRuleException(string rule, string reason) 
        : base($"Regra de negócio violada ({rule}): {reason}")
    {
    }
}

File: Java/src/main/java/com/foodcall/domain/exceptions/BusinessRuleException.java
package com.foodcall.domain.exceptions;

public class BusinessRuleException extends DomainException {
    public BusinessRuleException(String message) {
        super(message);
    }

    public BusinessRuleException(String rule, String reason){
        super(String.format("Regra de negócio violada (%s): %s", rule, reason));
    }
    
}
